src/metabase/driver/generic_sql/query_processor.clj

```diff
@@ -285,43 +288,74 @@
         (recur korma-form more)
         korma-form))))
 
-(defn- do-with-timezone [driver timezone f]
-  (log/debug (u/format-color 'blue (sql/set-timezone-sql driver)))
-  (try (kdb/transaction (k/exec-raw [(sql/set-timezone-sql driver) [timezone]])
-                        (f))
-       (catch Throwable e
-         (log/error (u/format-color 'red "Failed to set timezone:\n%s"
-                      (with-out-str (jdbc/print-sql-exception-chain e))))
-         (f))))
-
-(defn- do-with-try-catch [f]
-  (try
-    (f)
-    (catch java.sql.SQLException e
-      (jdbc/print-sql-exception-chain e)
-      (let [^String message (or (->> (.getMessage e)                       ; error message comes back like "Error message ... [status-code]" sometimes
-                                     (re-find  #"(?s)(^.*)\s+\[[\d-]+\]$") ; status code isn't useful and makes unit tests hard to write so strip it off
-                                     second)                               ; (?s) = Pattern.DOTALL - tell regex `.` to match newline characters as well
-                                (.getMessage e))]
-        (throw (Exception. message))))))
-
 (defn build-korma-form
   "Build the korma form we will call `k/exec` on."
   [driver {inner-query :query :as outer-query} entity]
   (binding [*query* outer-query]
       (apply-clauses driver (k/select* entity) inner-query)))
 
-(defn process-mbql
-  "Convert QUERY into a korma `select` form, execute it, and annotate the results."
-  [driver {{:keys [source-table]} :query, database :database, settings :settings, :as outer-query}]
-  (let [timezone     (:report-timezone settings)
-        entity       ((resolve 'metabase.driver.generic-sql/korma-entity) database source-table)
-        korma-form   (build-korma-form driver outer-query entity)
-        f             (partial k/exec korma-form)
-        f             (fn []
-                        (kdb/with-db (:db entity)
-                          (if (seq timezone)
-                            (do-with-timezone driver timezone f)
-                            (f))))]
-    (log-korma-form korma-form)
-    (do-with-try-catch f)))
+
+(def ^:private formatter (tf/formatter "YYYY-MM-dd" (t/default-time-zone)))
+
+;; TODO: we may need to make this part of ISQLDriver interface and let other dbs override as needed
+;; Boolean, Long, Double, String, Timestamp
+(defn- sql-param
+  "Format a single value properly for inclusion in a SQL statement"
+  [param]
+  ;; TODO: we need to handle sql date/time/timestamp values here (do actual jdbc drivers have functions for this?)
+  (cond
+    (u/is-temporal? param) (str "'" (tf/unparse formatter (tc/from-sql-time param)) "'")
+    (string? param)        (str "'" param "'")
```

Comment:
same note as above.  this code isn't active and i'll just delete it and let this be handled in the honeysql branch.